pyo3-macros/src/lib.rs

```diff
@@ -55,26 +67,152 @@ pub fn pyproto(_: TokenStream, input: TokenStream) -> TokenStream {
     .into()
 }
 
+/// A proc macro used to expose Rust structs as Python objects.
+///
+/// `#[pyclass]` accepts the following [parameters][2]:
+///
+/// |  Parameter  |  Description |
+/// | :-  | :- |
+/// | `name = "python_name"` | Sets the name that Python sees this class as. Defaults to the name of the Rust struct. |
+/// | `freelist = N`  |  Implements a [free list][10] of size N. This can improve performance for types that are often created and deleted in a row. Profile your code to see whether `freelist` is right for you.  |
+/// | `gc`  | Participate in Python's [garbage collection][5]. Required if your type contains references to other Python objects. If you don't (or incorrectly) implement this, contained Python objects may be hidden from Python's garbage collector and you may leak memory. Note that leaking memory, while undesirable, [is safe behavior][7].|
+/// | `weakref` | Allows this class to be [weakly referenceable][6]. |
+/// | `extends = BaseType`  | Use a custom baseclass. Defaults to [`PyAny`][4] |
+/// | `subclass` | Allows Python classes to inherit from this class.  |
+/// | `unsendable`<!-- hack to stop the column from wrapping -->&#160;&#160;&#160;&#160;&#160;&#160;&#160;&#160;&#160;&#160;&#160;&#160;&#160;&#160;&#160;&#160;&#160;&#160;&#160;&#160;&#160;&#160;&#160;&#160;&#160;&#160;&#160;&#160;&#160;&#160;&#160;&#160; | Required if your struct is not [`Send`][3]. Rather than using `unsendable`, consider implementing your struct in a threadsafe way by e.g. substituting [`Rc`][8] with [`Arc`][9]. By using `unsendable`, your class will panic when accessed by another thread.|
```

Comment:
Can you explain how this hack works? 🧠

Reply:
They're whitespaces that do not permit wrapping.

If something tries minimizing the table column, "ham spam eggs" can wrap around their spaces, but "ham&#160;spam&#160;eggs" (`ham&#160;spam&#160;eggs`) will not.

Reply:
Ah, understood.

I think it's slightly more intuitive (although admittedly more ugly) to do something like 

```
`<span style="white-space: pre">`name = "python_name"`</span>`
```

on the rows that we don't want to wrap, but I'll do that in a separate PR.